controllers/memcached/memcached_controller.go

```diff
@@ -185,9 +207,37 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return rbacResult, nil
 	}
 
+	// Hash of all resources that may cause a service restart
+	inputHashEnv := make(map[string]env.Setter)
+
+	// Check and hash inputs
+	var certHash, caHash string
+	specTLS := &instance.Spec.TLS
+	if err == nil && specTLS.Enabled() {
+		certHash, _, err = specTLS.GenericService.ValidateCertSecret(ctx, helper, instance.Namespace)
+		inputHashEnv["Cert"] = env.SetValue(certHash)
+	}
+	if err == nil && specTLS.Ca.CaBundleSecretName != "" {
+		caName := types.NamespacedName{
+			Name:      specTLS.Ca.CaBundleSecretName,
+			Namespace: instance.Namespace,
+		}
+		caHash, _, err = tls.ValidateCACertSecret(ctx, helper.GetClient(), caName)
```

Comment:
just realized this with how errors are handled here. the check for `err != nil` happens bellow and we still update the hash when the error happens in next line. in case of an existing deployment where everything is ok, if there is an error returned, e.g. a temporary issue with reading from etcd, we can see the hash flipping as it will be temporary at least to be set to "". since you use `certHash` and `caHash` vars we could just update the `inputHashEnv` after the `err != nil` check.

Reply:
same applies to redis https://github.com/openstack-k8s-operators/infra-operator/blob/main/controllers/redis/redis_controller.go#L220

Reply:
Right I overlook the fact that the defer function will kick in even when we return an error. Should be fixed now.